### DISCUSSION TITLE
How to date trees with date ranges for calibration "points";  confidence intervals

### DISCUSSION THREAD

**Discussion by bonisko** [iqtree/iqtree2#313] [General]
Hi everyone!

I have been using IQTree 2.3.4 for Windows with great success!  Thanks!

However I have been trying to compare dated tree results to those others have obtained with MCMCTree.

I can put median dates in Date_File.txt AOK, but is there a way to input a date range for calibrated nodes?

Also, I have tried the "--date-ci NUM" switch, and get a confidence interval OK in the output file *.timetree.lsd  But I really do not know what I am doing.  Is the method published yet?  Hope so!

Thx again!

Bruce

**Comment by roblanf** [comment]
just tagging @tothuhien

**Comment by tothuhien** [comment]
Hi Bruce,
Thanks for your question. You can find more details in this [documentation](http://www.iqtree.org/doc/Dating) 
Briefly, yes you can give a range for input date, for example: 
`taxon1,taxon2 2000:2002`
means that the node, which is most recent common ancestor of taxon1 and taxon2, has date between 2000 and 2002.
For the confidence interval, when you do "--date-ci NUM", it will
`resample branch lengths NUM times to infer the confidence intervals. Note that this is not bootstrap and the method is much faster but unpublished. Roughly speaking, it is based on a mixture of Poisson and lognormal distributions for a relaxed clock model.` The method has not been published yet, for some reasons the paper has been delayed. But we've tested it on lots of data and it gives pretty much similar results as bootstrap methods, but faster.
Best,
Hien

**Reply by bonisko** [reply]
Hi Hien!

Thank you for your helpful reply.  I have edited my date file, and repeated the analysis, but now get the following error:

![image](https://github.com/user-attachments/assets/4cb62468-5b7c-4bd2-b3cb-da775c70f3ed)

Perhaps I need to root my tree?  This is easy since about 1/2 my sequences are bacterial and the others are archaeal.

Thanks again!

Bruce

**Reply by bonisko** [reply]
PS Apologies if I should have started a new thread.  Also, I can attach files if anyone is interested in helping.  :-)

**Comment by tothuhien** [comment]
Hi Bruce,
In your date file, does it contain only dates of internal nodes or is there any tip? How do you specify dates of the tips? If all the tips have date 0, you can add `--date-tip 0` . 
Best,
Hien

**Comment by bonisko** [comment]
Hi Hien,

Rats!  I tried your idea, but got the same error.

Might I send you my alignment and date file?

Thanks again!  Much appreciated!

Bruce

**Comment by tothuhien** [comment]
Hi, so it should be the problem that the input dates are not enough. You can send me the input files either here or via email tothuhien@gmail.com.
I will have a look.
Hien

**Comment by bonisko** [comment]
Thanks for your help!
